### PR TITLE
Post hawk 565 fix

### LINKED ISSUE #565: [MSSQL 2012] A column has been specified more than once in the order by list
| Question       | Answer
|----------------|-------------------------------
| Bundle version | composer show sonata-project/*
> sonata-project/admin-bundle              3.1.0 The missing Symfony Admin G...
> sonata-project/block-bundle              3.0.0 Symfony SonataBlockBundle
> sonata-project/cache                     1.0.7 Cache library
> sonata-project/core-bundle               3.0.0 Symfony SonataCoreBundle
> sonata-project/doctrine-orm-admin-bundle 3.0.1 Symfony Sonata / Integrate ...
> sonata-project/exporter                  1.4.1 Lightweight Exporter library
> sonata-project/formatter-bundle          3.0.1 Symfony SonataFormatterBundle

| Symfony version| composer show symfony/*
> symfony/assetic-bundle     v2.8.0  Integrates Assetic into Symfony2
> symfony/monolog-bundle     2.11.1  Symfony MonologBundle
> symfony/polyfill-apcu      v1.2.0  Symfony polyfill backporting apcu_* fun...
> symfony/polyfill-intl-icu  v1.2.0  Symfony polyfill for intl's ICU-related...
> symfony/polyfill-mbstring  v1.2.0  Symfony polyfill for the Mbstring exten...
> symfony/polyfill-php54     v1.2.0  Symfony polyfill backporting some PHP 5...
> symfony/polyfill-php55     v1.2.0  Symfony polyfill backporting some PHP 5...
> symfony/polyfill-php56     v1.2.0  Symfony polyfill backporting some PHP 5...
> symfony/polyfill-php70     v1.2.0  Symfony polyfill backporting some PHP 7...
> symfony/polyfill-util      v1.2.0  Symfony utilities for portability of PH...
> symfony/security-acl       v3.0.0  Symfony Security Component - ACL (Acces...
> symfony/swiftmailer-bundle v2.3.11 Symfony SwiftmailerBundle
> symfony/symfony            v2.8.6  The Symfony PHP framework
> symfony/var-dumper         v3.0.6  Symfony mechanism for exploring and dum...

| php version    | php --version
> 7.0.5
# Error message

```
SQLSTATE[42000]: [Microsoft][ODBC Driver 11 for SQL Server][SQL Server]A column has been specified more than once in the order by list. Columns in the order by list must be unique.") in SonataAdminBundle:CRUD:base_list.html.twig at line 29.
```

Hi!
This commit 876f09853f3c1ea7699e0e567860d163961932d2 added default sort by. But if method getSortBy() return sorting, "order by" clause will be add twice. This throws critical error in mssql. Maybe move this part in "else" section execute() method?

### PR DESCRIPTION
**Pull request by greg0ire**
Closes #565
Replaces #566

### Changelog

```markdown
### Fixed
- Avoid duplicate field in ORDER clause
```
### Subject
Fix issue #565 "A column has been specified more than once in the order by list"

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "require-dev": {
         "knplabs/knp-menu-bundle": "^2.1.1",
         "simplethings/entity-audit-bundle": "~0.1",
+        "sllh/php-cs-fixer-styleci-bridge": "^2.0",
```
Comment (OskarStark) [inline composer.json]: why=

File: Tests/Datagrid/ProxyQueryTest.php [hunk 2]
```diff
@@ -99,10 +100,19 @@ public function testGetFixedQueryBuilder($class, $alias, $id, $expectedId, $valu
...
+                $parts = array(
+                    'from' => array(new From($class, $alias)),
+                    'orderBy' => array(new OrderBy('whatever', 'DESC')),
+                );
+
+                return $parts[$part];
+            }));
+        $qb->expects($this->once())
+            ->method('addOrderBy')
+            ->with("$alias.$id", null);
```
Comment (OskarStark) [inline Tests/Datagrid/ProxyQueryTest.php]: please use `'` instead of `"`

**Comment by OskarStark** [comment]
LGTM 👍

File: composer.json (hunk 1 above)
Comment (greg0ire) [inline composer.json]: You can't fix cs locally without this package

File: Tests/Datagrid/ProxyQueryTest.php (hunk 2 above)
Comment (greg0ire) [inline Tests/Datagrid/ProxyQueryTest.php]: `$alias . '.' . $id` is a bit harder to read, don't you think? So is `sprintf( '%s.%s', $alias, $id)`, no?

File: composer.json (hunk 1 above)
Comment (OskarStark) [inline composer.json]: ok, but no extra PR?

File: Tests/Datagrid/ProxyQueryTest.php (hunk 2 above)
Comment (OskarStark) [inline Tests/Datagrid/ProxyQueryTest.php]: 👍  for sprintf

you are right, its hard to read, but we should not use the `"`, i dont like them 😄

File: composer.json (hunk 1 above)
Comment (greg0ire) [inline composer.json]: An extra commit is quite enough IMO, especially for a change that small.

File: Tests/Datagrid/ProxyQueryTest.php (hunk 2 above)
Comment (greg0ire) [inline Tests/Datagrid/ProxyQueryTest.php]: > i dont like them :smile:

Why not ?

Comment (OskarStark) [inline Tests/Datagrid/ProxyQueryTest.php]: i donÄt know, in general its not so readable to me, but in this case it might be fine!

Lets keep it

**Comment by OskarStark** [comment]
@Soullivaneuh can you give us a final review an push the button if you agree?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,4 +5,4 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 ## [3.0.1](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/compare/3.0.0...3.0.1) - 2016-05-22
 ### Fixed
 - Added missing default sort by primary key(s).
-- Allow non integer/string types as identifier (ex. uuid).
+- Allow non integer/string types as identifier (ex. uuid).
```
Comment (Soullivaneuh) [inline CHANGELOG.md]: Remove this from this PR.

Or at least, do it on another commit.

File: composer.json (hunk 1 above)
Comment (Soullivaneuh) [inline composer.json]: > An extra commit is quite enough IMO, especially for a change that small.

This is not a matter of taste. :-)

https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/CONTRIBUTING.md#the-content

Several commit if this is more or less related to the PR, why not. For the rest, it would be great to have separated PR later.

File: CHANGELOG.md (hunk 3 above)
Comment (greg0ire) [inline CHANGELOG.md]: Oh right! Forgot that.

Comment (greg0ire) [inline CHANGELOG.md]: @Soullivaneuh fixed

File: Tests/Datagrid/ProxyQueryTest.php (hunk 2 above)
Comment (greg0ire) [inline Tests/Datagrid/ProxyQueryTest.php]: Thanks for being understanding, I was like you before, because I "heard" if was bad, then lately I realized it is bad in some cases, and in some very simple other cases, it's better because it is way more readable. It's a bit like Yoda conditions if you will : a safe-guard that protects you from errors you don't make anymore, but bit you hard one time and made you decide to embrace it. But now that we have code review, I think we can avoid these mistakes easily.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,4 +5,4 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 ## [3.0.1](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/compare/3.0.0...3.0.1) - 2016-05-22
 ### Fixed
 - Added missing default sort by primary key(s).
-- Allow non integer/string types as identifier (ex. uuid). 
+- Allow non integer/string types as identifier (ex. uuid).
```
Comment (Soullivaneuh) [inline CHANGELOG.md]: This is introduced by this PR commit: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/567/commits/1001034f4b2a41ff40c65c4a730df67508f97740#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR8

Just squash them and get rid of this.

Comment (greg0ire) [inline CHANGELOG.md]: O_o I don't understand how this is possible, but ok…

Comment (OskarStark) [inline CHANGELOG.md]: Add another commit which adds the whitespace and squash it ;-)

**Comment by Soullivaneuh** [comment]
Looks good.

@greg0ire If you are sure of the patch, feel free to merge. :+1:

**Comment by greg0ire** [comment]
Thanks @postHawk !
